talk/tools/compiling.tex

```diff
@@ -109,6 +109,49 @@
   \end{minted}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitle{CMake}
+  \begin{block}{}
+    \begin{itemize}
+      \item a cross-platform meta build system
+      \item generates platform-specific build systems
+      \item see also this \href{https://www.youtube.com/watch?v=eC9-iRN2b04}{basic} and \href{https://www.youtube.com/watch?v=bsXLMQ6WgIk}{detailed} talks
+    \end{itemize}
+  \end{block}
+  \begin{block}{Example CMakeLists.txt}
+    \begin{minted}[linenos=true,autogobble]{cmake}
+      cmake_minimum_required(VERSION 3.18)
+      project(hello CXX)
+
+      find_package(ZLIB REQUIRED) # for external libs
+
+      add_executable(hello main.cpp util.h util.cpp)
+      target_compile_features(hello PUBLIC cxx_std_17)
+      target_link_libraries(hello PUBLIC ZLIB::ZLIB)
+    \end{minted}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{CMake - Building}
+  \begin{block}{Building a CMake-based project}
+    Start in the directory with the top-level \texttt{CMakeLists.txt}:
+    \begin{minted}[linenos=true,autogobble]{bash}
+      mkdir build # will contain all build-related files
```

Comment:
That is where I usually create build folders (top level inside the working copy), but I have also seen people create the build folder completely outside the working copy. What's your take on this?

Reply:
I do both. Imagine the case where you have code on a network file system and object files on an ssd, space problems on one device, stuff you want to throw away in `/tmp`, tons of things.
I think it's fine to show build folder in source folder, but we should *say* in the lecture that you can put it anywhere.